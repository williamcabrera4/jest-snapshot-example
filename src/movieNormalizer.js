const movieNormalizer = (movieData) => ({
  title: movieData.Title,
  description: movieData.Plot,
  language: movieData.Language,
  writers: movieData.Writer.split(','),
  actors: movieData.Actors.split(','),
  genres: movieData.Genre.split(','),
});

export default movieNormalizer;
