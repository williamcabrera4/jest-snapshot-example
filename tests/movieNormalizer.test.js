import movieNormalizer from '../src/movieNormalizer';
import movie from './movie.json';

describe('Movie Normalizer', () => {

  test('should return a normalized object from the JSON API schema', () => {
    const normalizedMovie = movieNormalizer(movie);
    expect(normalizedMovie).toMatchSnapshot();
  });
});