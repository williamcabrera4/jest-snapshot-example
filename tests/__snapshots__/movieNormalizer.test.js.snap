// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Normalizer should return a normalized object from the JSON API schema 1`] = `
Object {
  "actors": Array [
    "Peter Dinklage",
    " Lena Headey",
    " Emilia Clarke",
    " Kit Harington",
  ],
  "description": "Nine noble families fight for control over the mythical lands of Westeros, while a forgotten race returns after being dormant for thousands of years.",
  "genres": Array [
    "Adventure",
    " Drama",
    " Fantasy",
  ],
  "language": "English",
  "title": "Game of Thrones",
  "writers": Array [
    "David Benioff",
    " D.B. Weiss",
  ],
}
`;
